test/e2e/util/hypershift_framework.go

```diff
@@ -69,15 +74,24 @@ func NewHypershiftTest(t *testing.T, ctx context.Context, test hypershiftTestFun
 	}
 }
 
+// NewHypershiftTestWithManualTeardown is used when the test needs to control when to trigger the teardown. Please note
+// that teardown will be triggered anyway in case of errors (e.g. if creating a HC fails)
+func NewHypershiftTestWithManualTeardown(t *testing.T, ctx context.Context, test hypershiftTestFunc) *hypershiftTest {
+	hst := NewHypershiftTest(t, ctx, test)
+	hst.manualTeardown = true
```

Comment:
I don't think we want to add the option to skip teardown, it can be easily be misused and leave resources in the cloud.

For the purpose of this task, I don't really think you need to use the hypershift e2e framework to create the cluster. All the validations ran by the framework are not important for the upgrade test and will unnecessarily increase the execution time.

Reply:
- Who exactly would misuse this? I don't think this is a public API nor a customer used one, correct? 
- I prefer reusing the same logic that validates that a created test HC is healthy. That way if one HC health invariant changes in one place, it changes in another.
- This test is expected to run solo in parallel to the existing e2e tests, so it won't affect the overall pipeline execution time.